pkg/api/user.go

```diff
@@ -33,7 +33,7 @@ func getUserUserProfile(userId int64) Response {
 
 // GET /api/users/loginoremail/:loginOrEmail
 func GetUserByLoginOrEmail(c *middleware.Context) Response {
-	query := m.GetUserByLoginQuery{LoginOrEmail: c.Params(":loginOrEmail")}
+	query := m.GetUserByLoginQuery{LoginOrEmail: c.QueryEscape("loginOrEmail")}
```

Comment:
> For simple cases it returns correct results 

What case does it not return correct results? 

> which takes also into account post requests for forms. 

But this is a GET request so not sure that applies here. 

1) special charachers are URL encoded by user who uses the API, decoded automatically by golang http library
2) Grafana is not passing any personal information as this is an external API call (That grafana UI is not using). And there is no inherent security difference between POST and GET if the call is unencrypted